File: src/main/java/org/mifos/fineract/services/PovertyLineApi.java
package org.mifos.fineract.services;

import retrofit2.http.GET;
import retrofit2.http.Headers;
import rx.Observable;

public interface PovertyLineApi {
    /**
     * @param ppiName (required)
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("povertyLine/{ppiName}")
    Observable<String> retrieveAll(
            @retrofit2.http.Path("ppiName") String ppiName
    );

    /**
     * @param ppiName      (required)
     * @param likelihoodId (required)
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("povertyLine/{ppiName}/{likelihoodId}")
    Observable<String> retrieveAll_0(
            @retrofit2.http.Path("ppiName") String ppiName, @retrofit2.http.Path("likelihoodId") Long likelihoodId
    );

}

File: src/main/kotlin/org/openapitools/client/apis/PovertyLineApi.kt
package org.openapitools.client.apis

import retrofit2.http.GET
import retrofit2.http.Path


interface PovertyLineApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param ppiName
     * @return [kotlin.String]
     */
    @GET("v1/povertyLine/{ppiName}")
    suspend fun retrieveAll12(@Path("ppiName") ppiName: String): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param ppiName
     * @param likelihoodId
     * @return [kotlin.String]
     */
    @GET("v1/povertyLine/{ppiName}/{likelihoodId}")
    suspend fun retrieveAll13(
        @Path("ppiName") ppiName: String,
        @Path("likelihoodId") likelihoodId: Long
    ): String

}
